packages/rich-text/src/component/use-anchor-ref.js

```diff
@@ -44,7 +50,12 @@ export function useAnchorRef( { ref, value, settings = {} } ) {
 		const range = selection.getRangeAt( 0 );
 
 		if ( ! activeFormat ) {
-			return range;
+			return {
+				ownerDocument: range.startContainer.ownerDocument,
+				getBoundingClientRect() {
+					return range.getBoundingClientRect();
+				},
+			};
```

Comment:
Main change: this hook now returns a `VirtualElement` instead of a `Range`.

A `VirtualElement` is an element with the `ownerDocument` and `getBoundingClientRect()` properties, and is used by `floating-ui` to determine an anchor when there's not an actual DOM `Element` available.

Couple of notes:

- for the `getBoundingClientRect()`, I decided to keep the same logic as currently in the `Popover` (i.e, just call `getBoundingClientRect()` on the range). Alternatively, we could use the [getRectangleFromRange](https://github.com/WordPress/gutenberg/blob/60750ceb29f209e8d118e2ee5727094a50be29a4/packages/dom/src/dom/get-rectangle-from-range.js) utility, which was used in the `Popover` component before the refactor happened in #40740
- for the `ownerDocument`, I kept [the same logic as the one from the `Popover` component](https://github.com/WordPress/gutenberg/blob/4e7508b74e08f03f61b56ecafa6d57fdfedfb3e7/packages/components/src/popover/utils.js#L135-L136)